test/molecular_hamiltonian_tests.jl

```diff
@@ -0,0 +1,44 @@
+using Rimu
+using ElemCo
+using Test
+using Rimu.InterfaceTests: test_observable_interface, test_operator_interface,
+    test_hamiltonian_interface, test_hamiltonian_structure
+
+
+@testset "MolecularHamitonian" begin
+    molecules = ("h2", "h2o")
+    error = 1e-10
+    @testset "Basic interface functionality" begin
+        fcidump = string(@__DIR__, "/examples/h2.FCIDUMP")
+        h = MolecularHamiltonian(fcidump)
+        test_hamiltonian_interface(h)
+        test_hamiltonian_structure(h)
+    end
+
+    @testset "Hartree-Fock ground state energy" begin
+        for molecule in molecules
+            fcidump = string(@__DIR__, "/examples/$(molecule).FCIDUMP")
+            ref_hf_ground_energy = @bohf
+            h = MolecularHamiltonian(fcidump)
+            a = starting_address(h)
+            c = operator_column(h, a)
+            @test diagonal_element(c) ≈ ref_hf_ground_energy atol = error
+        end
+    end
+
+    @testset "Off-diagonal states and energies" begin
+        h2_fci_matrix = [
+            -1.117008268 0 0 0.1809125853;
+            0 -0.3437220243 0.1809125853 0;
+            0 0.1809125853 -0.3437220243 0;
+            0.1809125853 0 0 0.4746278719
+        ]
+        fcidump = string(@__DIR__, "/examples/h2.FCIDUMP")
```

Comment:
```suggestion
        fcidump = joinpath(@__DIR__, "/examples/h2.FCIDUMP")
```
What you wrote is not wrong, but using `joinpath` is preferred when dealing with paths. It makes your intention clearer and I think it handles path separators on Windows for you.